api/dbadapters/bigquery.ts

```diff
@@ -190,6 +194,28 @@ export class BigQueryDbAdapter implements IDbAdapter {
 
   public async close() {}
 
+  public async persistedStateMetadata(
+    compiledGraph: dataform.ICompiledGraph
+  ): Promise<dataform.IPersistedTableMetadata[]> {
+    const adapter = adapters.create(
+      compiledGraph.projectConfig,
+      compiledGraph.dataformCoreVersion || "1.0.0"
+    );
+    try {
+      const { rows } = await this.runQuery(
+        `SELECT * FROM ${adapter.resolveTarget(STATE_PERSIST_TABLE_TARGET)}`,
+        5000 // not expecting to have more dataset than this
+      );
+      const peristedMetadata = rows.map(row => {
+        const encodedProto = Buffer.from(row.proto, "base64");
+        return dataform.PersistedTableMetadata.decode(encodedProto);
+      });
+      return peristedMetadata || [];
+    } catch (err) {
+      return [];
```

Comment:
it feels pretty unsafe to be doing this at this level! perhaps in the runner this is OK, but inside the DB adapter this seems a bit dangerous.